src/main/java/uk/ac/ebi/embl/converter/cli/Main.java

```diff
@@ -0,0 +1,163 @@
+import ch.qos.logback.classic.LoggerContext;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+import picocli.CommandLine.*;
+import picocli.CommandLine.Model.CommandSpec;
+import uk.ac.ebi.embl.converter.fftogff3.FFToGff3Converter;
+import uk.ac.ebi.embl.converter.fftogff3.FFtoGFF3ConversionError;
+import uk.ac.ebi.embl.converter.gff3toff.Gff3ToFFConverter;
+
+@Command(
+        name = "gff3tools",
+        subcommands = {CommandConversion.class, CommandLine.HelpCommand.class},
+        description = "Utility to convert and validate gff3 files")
+public class Main {
+    @Spec
+    CommandSpec spec;
+
+    private static final Logger LOG = LoggerFactory.getLogger(Main.class);
+
+    public static void main(String[] args) {
+        int exitCode = -1;
+        try {
+            exitCode = new CommandLine(new Main()).execute(args);
+        } catch (Exception e) {
+            LOG.error(e.getMessage(), e);
+        }
+        System.exit(exitCode);
+    }
+}
+
+// Using pandoc CLI interface conventions
+@Command(name = "conversion", description = "Performs format conversions to or from gff3")
+class CommandConversion implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(CommandConversion.class);
+
+    public enum FileFormat {
+        ff,
+        gff3
+    }
+
+    @Option(names = "-f", description = "The type of the file to be converted")
+    public FileFormat fromFileType;
+
+    @Option(names = "-t", description = "The type of the file to convert to")
+    public FileFormat toFileType;
+
+    @Parameters(paramLabel = "[input-file]", defaultValue = "", showDefaultValue = Help.Visibility.NEVER)
+    public Path inputFilePath;
+
+    @Parameters(paramLabel = "[output-file]", defaultValue = "", showDefaultValue = Help.Visibility.NEVER)
+    public Path outputFilePath;
+
+    @Override
+    public void run() {
+        Path emptyPath = Path.of("");
+        if (fromFileType == null) {
+            if (!inputFilePath.equals(emptyPath)) {
+                String fileExtension = getFileExtension(inputFilePath);
+                if (fileExtension != null) {
+                    try {
+                        fromFileType = FileFormat.valueOf(fileExtension);
+                    } catch (IllegalArgumentException e) {
+                        throw new Error("Unrecognized file format: " + fileExtension
+                                + " use the -f option to specify the format manually or update the file extension");
+                    }
+                } else {
+                    throw new Error(
+                            "No file extension present, use the -f option to specify the format manually or set the file extension");
+                }
+            } else {
+                throw new Error("When using stdin -f must be specified");
+            }
+        }
```

Comment:
This code is repetitive for input and output file we could extract to a method and call for both input and output file